src/Tempest/Console/src/GenericConsole.php

```diff
@@ -197,17 +226,38 @@ public function progressBar(iterable $data, Closure $handler): array
         return $this->component(new ProgressBarComponent($data, $handler));
     }
 
-    public function search(string $label, Closure $search): mixed
+    public function search(string $label, Closure $search, ?string $default = null): mixed
     {
-        return $this->component(new SearchComponent($label, $search));
+        return $this->component(new SearchComponent($label, $search, $default));
     }
 
-    private function interactiveSupported(): bool
+    public function supportsTty(): bool
```

Comment:
Ah but I see it's just a rename of `interactiveSupported`. I don't remember why I built it like this originally… Probably because I wanted to take a shortcut while hacking on interactive support, and never remembered to properly refactor it. 

It's not a problem for now, let's keep it, but it might be good to refactor it at a later point.

Reply:
I'll try to refactor it in my upcoming prompt overhaul PR 👍